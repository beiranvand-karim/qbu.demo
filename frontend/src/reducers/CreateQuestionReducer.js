import {
   CREATE_QUESTION_BEGIN,
   CREATE_QUESTION_SUCCESS,
   CREATE_QUESTION_ERROR
} from '../actions/CreateQuestionActions'

const initialState = {
   loading: false,
   error: null,
   data: null
};

export default function CreateQuestionReducer(state = initialState, action) {
   switch (action.type) {
      case CREATE_QUESTION_BEGIN:
         return {
            ...state,
            loading: true,
            error: null,
            data: null
         };

      case CREATE_QUESTION_SUCCESS:
         return {
            ...state,
            loading: false,
            error: null,
            data: state.payload
         };

      case CREATE_QUESTION_ERROR:

         return {
            ...state,
            loading: false,
            error: state.error,
            data: null
         };

      default:
         return state;
   }
}
