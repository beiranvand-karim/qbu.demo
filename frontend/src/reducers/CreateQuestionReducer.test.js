import {
   CREATE_QUESTION_SUCCESS,
   CREATE_QUESTION_ERROR,
   CREATE_QUESTION_BEGIN
} from '../actions/CreateQuestionActions'
import CreateQuestionReducer from './CreateQuestionReducer'

describe('CreateQuestionReducer', () => {

   it('should matches the initial state', () => {
      const action = { type: 'DUMMY_ACTION' };
      expect(CreateQuestionReducer(undefined, action)).toMatchSnapshot()
   });

   it('should matches the create question begin state', () => {
      const action = { type: CREATE_QUESTION_BEGIN};
      expect(CreateQuestionReducer(undefined, action)).toMatchSnapshot()
   });

   it('should matches the create question successful state', () => {
      const action = {
         type: CREATE_QUESTION_SUCCESS,
         payload: 'cc0f80d7-ca5c-4c97-aa45-8ee02a5b80de'
      };
      expect(CreateQuestionReducer(undefined, action)).toMatchSnapshot()
   });

   it('should matches the create question error state', () => {
      const action = {
         type: CREATE_QUESTION_ERROR,
         error: 'this is a error'
      };
      expect(CreateQuestionReducer(undefined, action)).toMatchSnapshot()
   });

});
