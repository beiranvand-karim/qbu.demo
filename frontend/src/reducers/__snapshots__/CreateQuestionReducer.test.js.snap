// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateQuestionReducer should matches the create question begin state 1`] = `
Object {
  "data": null,
  "error": null,
  "loading": true,
}
`;

exports[`CreateQuestionReducer should matches the create question error state 1`] = `
Object {
  "data": null,
  "error": "this is a error",
  "loading": false,
}
`;

exports[`CreateQuestionReducer should matches the create question successful state 1`] = `
Object {
  "data": "cc0f80d7-ca5c-4c97-aa45-8ee02a5b80de",
  "error": null,
  "loading": false,
}
`;

exports[`CreateQuestionReducer should matches the initial state 1`] = `
Object {
  "data": null,
  "error": null,
  "loading": false,
}
`;
